relay_messages.go

```diff
@@ -110,19 +112,16 @@ func newLazyCallReq(f *Frame) lazyCallReq {
 
 	serviceLen := f.Payload[_serviceLenIndex]
 	// nh:1 (hk~1 hv~1){nh}
-	headerStart := _serviceLenIndex + 1 /* length byte */ + serviceLen
-	numHeaders := int(f.Payload[headerStart])
-	cur := int(headerStart) + 1
-	for i := 0; i < numHeaders; i++ {
-		keyLen := int(f.Payload[cur])
-		cur++
-		key := f.Payload[cur : cur+keyLen]
-		cur += keyLen
+	headerStartIndex := _serviceLenIndex + 1 /* length byte */ + serviceLen
+
+	rbuf := typed.NewReadBuffer(f.SizedPayload()[headerStartIndex:])
```

Comment:
heads up that when using this, we should be careful to check `rbuf.Err()`,
https://godoc.org/github.com/uber/tchannel-go/typed#ReadBuffer.Err

the other methods will set the error if there's not enough bytes to read etc, so we might not notice if we don't check that

Reply:
yea will need to think about this a little more, not sure if we should do granular checks or just a single check at the end

Reply:
I think a single check at the end is probably fine, since we can't really react to intermediate errors differently (in either case, it's an invalid frame and we'll reject it)